core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -95,99 +92,86 @@ public void addCustomNameResolver(CustomNameResolver customNameResolver) {
     }
 
 
-    public InetAddress resolveBindHostAddress(String bindHost) throws IOException {
-        return resolveBindHostAddress(bindHost, InetAddress.getLoopbackAddress().getHostAddress());
+    public InetAddress[] resolveBindHostAddress(String bindHost) throws IOException {
+        return resolveBindHostAddress(bindHost, "_local_");
     }
 
-    public InetAddress resolveBindHostAddress(String bindHost, String defaultValue2) throws IOException {
-        return resolveInetAddress(bindHost, settings.get(GLOBAL_NETWORK_BINDHOST_SETTING, settings.get(GLOBAL_NETWORK_HOST_SETTING)), defaultValue2);
+    private InetAddress[] resolveBindHostAddress(String bindHost, String defaultValue2) throws IOException {
+        /** TODO: move this leniency out */
+        if (bindHost == null) {
+            bindHost = settings.get(GLOBAL_NETWORK_BINDHOST_SETTING, settings.get(GLOBAL_NETWORK_HOST_SETTING));
+        }
+        if (bindHost == null) {
+            bindHost = defaultValue2;
```

Comment:
I have changes coming in a commit. Basically it was too hard for me to see the various fallbacks for the default case (null).